src/pages/Lending/Pool/components/Borrow/BorrowInput.tsx

```diff
@@ -275,6 +278,33 @@ export const BorrowInput = ({
     isEstimatedSweepFeesDataSuccess,
   ])
 
+  const { data: _isSmartContractAddress, isLoading: isAddressByteCodeLoading } = useQuery({
+    queryKey: [
+      'isSmartContractAddress',
+      {
+        userAddress: collateralAccountId
+          ? fromAccountId(collateralAccountId).account.toLowerCase()
+          : '',
+      },
+    ],
+    queryFn: () =>
+      isSmartContractAddress(
+        collateralAccountId ? fromAccountId(collateralAccountId).account.toLowerCase() : '',
+      ),
+  })
+
+  const disableSmartContractDeposit = useMemo(() => {
+    // Collateral AccountId still loading - disable confirm
+    if (!collateralAccountId) return true
+    if (!isAddress(fromAccountId(collateralAccountId).account)) return false
```

Comment:
could reduce to one `if (!isAddress(userAddress)` check with the memoized `userAddress` variable. Also, shouldn't we be returning `true` if it is not a valid address? would be more of a generic disable func as it disables button for invalid or smart contract address, but seems like what we want